example/lib/services/theme-provider.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+
+class ThemeProvider {
+  final FlutterSecureStorage _themeStorage = FlutterSecureStorage();
+  final _storageKey = "theme";
+
+  final ThemeData _lightTheme = ThemeData(
+    primaryColor: const Color(0xFF387463),
+    brightness: Brightness.light,
+  );
+
+  final ThemeData _darkTheme = ThemeData(
+    primaryColor: Colors.grey[900],
+    brightness: Brightness.dark,
+  );
+
+  final ValueNotifier<bool> _themeNotifier = ValueNotifier<bool>(false);
+
+  bool _isDarkMode = false;
+
+  bool get isDarkMode => _isDarkMode;
+  ThemeData get currentTheme => _isDarkMode ? _darkTheme : _lightTheme;
+
+  ValueNotifier<bool> get themeNotifier => _themeNotifier;
+
+  ThemeProvider() {
+    this._LoadThemeFromStorage();
+  }
+
+  void toggleTheme(bool val) {
+    _isDarkMode = val;
+    this._themeNotifier.value = this._isDarkMode;
+    this._updateThemeInStorage();
+  }
+
+  Future<void> _updateThemeInStorage() async {
+    await _themeStorage.write(key: this._storageKey, value: this._isDarkMode.toString());
+  }
+
+  Future<void> _LoadThemeFromStorage() async {
+    final storedValue = await this._themeStorage.read(key: this._storageKey);
+    if (storedValue != null) {
+      // ignore: sdk_version_since
+      this._isDarkMode = bool.parse(storedValue, caseSensitive: false);
```

Comment:
actually i am storing string of true or false right if you don't want bool.parse i can write my own parser still same right?

Reply:
you are using an ignore on the parse because it could fail if somebody with flutter version <3 runs this app it could fail. you should either set your min sdk version to 3 or don't use bool.parse. you should have a really good reason to write code with igonre